qubesadmin/tools/qvm_start_daemon.py

```diff
@@ -678,19 +731,68 @@ def on_domain_stopped(self, vm, _event, **_kwargs):
             return
         if xid != -1:
             self.cleanup_guid(xid)
+            self.cleanup_pacat_process(xid)
         if stubdom_xid != -1:
             self.cleanup_guid(stubdom_xid)
+            self.cleanup_pacat_process(stubdom_xid)
+
+    def on_property_audiovm_set(self, vm, _event, **_kwargs):
+        """Handler for catching event related to dynamic AudioVM set/unset"""
+        if vm.name not in self.xid_cache:
+            try:
+                self.xid_cache[vm.name] = vm.xid, vm.stubdom_xid
+            except qubesadmin.exc.QubesDaemonAccessError as e:
+                log.error("vm.name: failed to determine XID: %s", str(e))
+                return
+        xid, stubdom_xid = self.xid_cache[vm.name]
+        oldvalue = _kwargs.get("oldvalue", None)
+        newvalue = _kwargs.get("newvalue", None)
+        if oldvalue == self.app.local_name and newvalue == "None":
```

Comment:
Shouldn't it be `newvalue != self.app.local_name` ? Right now when I change audiovm from dom0 to sys-audio, the pacat process in dom0 remains running. Or is it intentional for more reliable switching back?

Reply:
But also, this means if I first switch from sys-audio to dom0 and then shutdown the qube, its pacat remains in sys-audio (it's terminated in dom0 only).

Reply:
> Shouldn't it be `newvalue != self.app.local_name` ? Right now when I change audiovm from dom0 to sys-audio, the pacat process in dom0 remains running. Or is it intentional for more reliable switching back?

No it's not intentional, I think it was at the very early work for this to handle all the cases in the same function without rethinking the whole after. I'll fix that indeed.